gravitee-am-identityprovider/gravitee-am-identityprovider-http/src/main/java/io/gravitee/am/identityprovider/http/utils/SanitizeUtils.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.identityprovider.http.utils;
+
+import io.gravitee.common.http.HttpHeader;
+import io.gravitee.common.http.HttpHeaders;
+import io.gravitee.common.http.MediaType;
+
+import java.util.List;
+
+/**
+ * @author Titouan COMPIEGNE (titouan.compiegne at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public final class SanitizeUtils {
+
+    /**
+     * Sanitize user credentials if a JSON body is sent
+     * @param credentials
+     * @param authenticationHttpHeaders
+     * @return
+     */
+    public static String sanitize(String credentials, List<HttpHeader> authenticationHttpHeaders) {
+        if (authenticationHttpHeaders == null || authenticationHttpHeaders.isEmpty()) {
+            return credentials;
+        }
+
+        if (!authenticationHttpHeaders
```

Comment:
if the content-type isn't specify, the body mat be a Json (this is the case with the default values, the body is a JSON but no headers are defined.) Do we have to consider JSON as default payload? If any, we have to test the presence of the Content-Type header separately.